src/init.c

```diff
@@ -406,8 +407,8 @@ DISPATCH_VTABLE_SUBCLASS_INSTANCE(queue_main, queue,
 	.do_type = DISPATCH_QUEUE_SERIAL_TYPE,
 	.do_kind = "main-queue",
 	.do_dispose = _dispatch_queue_dispose,
-	.do_invoke = _dispatch_queue_invoke,
 	.do_push = _dispatch_queue_push,
+	.do_invoke = _dispatch_queue_invoke,
```

Comment:
I just kept the order that was dominating. For real the better order is:

wakeup -> push -> invoke. (cause it's in that order it happens)